test/parallel/xyk-pallet.API.tokenTimeout.test.ts

```diff
@@ -0,0 +1,226 @@
+/*
+ *
+ * @group xyk
+ * @group parallel
+ */
+
+import { Keyring } from "@polkadot/api";
+import { getApi, initApi } from "../../utils/api";
+import { Assets } from "../../utils/Assets";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import {
+  ExtrinsicResult,
+  waitSudoOperataionSuccess,
+  waitSudoOperataionFail,
+} from "../../utils/eventListeners";
+import { BN } from "@mangata-finance/sdk";
+import { setupApi, setupUsers } from "../../utils/setup";
+import { Sudo } from "../../utils/sudo";
+import { sellAsset, updateTimeoutMetadata } from "../../utils/tx";
+import { getEventResultFromMangataTx } from "../../utils/txHandler";
+import { AssetWallet, User } from "../../utils/User";
+import { getEnvironmentRequiredVars } from "../../utils/utils";
+import { Xyk } from "../../utils/xyk";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(2500000);
+process.env.NODE_ENV = "test";
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+let testUser1: User;
+let sudo: User;
+let keyring: Keyring;
+let createdToken: BN;
+const defaultSwapValue = new BN(50000);
+const thresholdValue = new BN(30000);
+const defaultCurrencyValue = new BN(10000000);
+const defaultPoolVolumeValue = new BN(1000000);
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  keyring = new Keyring({ type: "sr25519" });
+
+  // setup users
+  sudo = new User(keyring, sudoUserName);
+
+  [testUser1] = setupUsers();
+
+  await setupApi();
+
+  createdToken = await Assets.issueAssetToUser(
+    sudo,
+    defaultCurrencyValue,
+    sudo
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(createdToken, sudo, defaultCurrencyValue),
+    Assets.mintToken(createdToken, testUser1, defaultCurrencyValue),
+    Assets.mintNative(testUser1),
+    Sudo.sudoAs(
+      sudo,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        defaultPoolVolumeValue,
+        createdToken,
+        defaultPoolVolumeValue
+      )
+    )
+  );
+});
+
+test("xyk-pallet-gassless GIVEN a token WHEN no token timeout is configured THEN fees are charged as normal", async () => {
+  const api = getApi();
+
+  const timeoutMetadata = await api.query.tokenTimeout.timeoutMetadata();
+
+  if (timeoutMetadata.isEmpty) {
```

Comment:
in slack we discussed that tokenTimout will  always exist for  timeoutMetadata. I added a few days ago in the global setup. Please  remove all the conditional tests